File: akito0107/alds1_4_c.js
'use strict';

var input = require('fs').readFileSync('/dev/stdin', 'utf8');
var lines = (input.trim()).split('\n');
lines.shift();
var hash = {};

lines.forEach(function(line) {
  var cmd = line.split(' ')[0];
  var key = line.split(' ')[1];
  if (cmd === 'insert') {
    hash[key] = true;
    return;
  }
  if (hash[key]) {
    console.log('yes');
  } else {
    console.log('no');
  }
});

//var util = require('util');
//waterfall([
//  function(next) {
//    lineReader(1, next);
//  }, function(line, next) {
//    var hash = {};
//    lineReader(parseInt(line[0]), function(line) {
//      var command = line.split(' ')[0];
//      var key = line.split(' ')[1];
//      if (command === 'insert') {
//        hash[key] = true;
//      } else {
//        if (hash[key]) {
//          util.print('yes\n');
//        } else {
//          util.print('no\n');
//        }
//      }
//    }, next);
//  }], function(result) {
//    process.exit(0);
//  });
//
//function strMap(str, map) {
//  var newStr = '';
//  for (var i = 0; i < str.length; i++) {
//    newStr += map(str.charAt(i), i);
//  }
//  return newStr;
//}
//
//function lineReader(limit, lineCb, cb) {
//  if (typeof cb === 'undefined') {
//    cb = lineCb;
//    lineCb = null;
//  }
//  var reader = require('readline').createInterface({
//    input: process.stdin,
//    output: process.stdout,
//  });
//  var lines = [];
//  var counter = 0;
//  reader.on('line', function(line) {
//    if (lineCb) {
//      lineCb(line);
//    }
//    lines.push(line);
//    ++counter;
//    if (counter === limit) {
//      reader.close();
//      return cb(lines);
//    }
//  });
//  reader.on('close', function() {});
//  reader.on('error', function(err) {
//    cb(err);
//  });
//}
//
//function waterfall(tasks, cb) {
//  var taskIndex = 0;
//  function nextTask(args) {
//    if (taskIndex === tasks.length)
//      return cb.apply(null, args);
//
//    var taskCallback = function() {
//      var _arguments = arguments;
//      nextTask(Object.keys(_arguments).reduce(function(memo, key) {
//        memo.push(_arguments[key]);
//        return memo;
//      }, []));
//    };
//    args.push(taskCallback);
//    var task = tasks[taskIndex++];
//    task.apply(null, args);
//  }
//  nextTask([]);
//}

File: sotetsuk/python/ALDS1_4_C.py
class Dict(object):

    m1 = 2 ** 24 - 1
    m2 = m1 - 1

    def __init__(self):
        self._dict = ["" for _ in range(self.m1)]

    def insert(self, k):
        i = 0

        while True:
            ix = self._h(k, i)
            if self._dict[ix] == "":
                self._dict[ix] = k
                return
            else:
                i += 1

    def find(self, k):
        i = 0

        while True:
            ix = self._h(k, i)
            if self._dict[ix] == k:
                return ix
            elif self._dict[ix] == "":
                return -1
            else:
                i += 1

    def _h(self, k, i):
        return self._h1(k) + i * self._h2(k)

    def _h1(self, k):
        return k % self.m1

    def _h2(self, k):
        return k % self.m2 + 1


def to_int(s):
    _s = s[::]

    ret = 0
    i = 1
    for e in _s:
        if e == 'A':
            ret += 1 * i
        elif e == 'C':
            ret += 2 * i
        elif e == 'G':
            ret += 3 * i
        elif e == 'T':
            ret += 4 * i
        i *= 4

    return ret


if __name__ == '__main__':
    n = int(input())

    d = Dict()

    for _ in range(n):
        cmd, val = input().split()

        if cmd == "insert":
            d.insert(to_int(val))
        else:
            ans = "yes" if d.find(to_int(val)) != -1 else "no"
            print(ans)
